public/components/common/hooks/useValueSuggestion/use-value-suggestion.ts

```diff
@@ -9,33 +9,51 @@
  *
  * Find more information about this on the LICENSE file.
  */
-import { useState, useEffect } from 'react';
-import { getDataPlugin } from '../../../kibana-services';
-import { useIndexPattern } from '.';
+
+import React, { useEffect, useState } from 'react';
+import { getDataPlugin } from '../../../../kibana-services';
+import { useIndexPattern } from '../../hooks';
 import { IFieldType, IIndexPattern } from 'src/plugins/data/public';
-import React from 'react';
 import {
   UI_ERROR_SEVERITIES,
   UIErrorLog,
   UIErrorSeverity,
   UILogLevel,
-} from '../../../react-services/error-orchestrator/types';
-import { UI_LOGGER_LEVELS } from '../../../../common/constants';
-import { getErrorOrchestrator } from '../../../react-services/common-services';
+} from '../../../../react-services/error-orchestrator/types';
+import { UI_LOGGER_LEVELS } from '../../../../../common/constants';
+import { getErrorOrchestrator } from '../../../../react-services/common-services';
+import { getCustomValueSuggestion } from './helpers/helper-value-suggestion';
 
-export interface IValueSuggestiions {
+export interface IValueSuggestion {
   suggestedValues: string[] | boolean[];
   isLoading: boolean;
   setQuery: React.Dispatch<React.SetStateAction<string>>;
 }
 
-export const useValueSuggestions = (filterField: string, type: 'string' | 'boolean' = 'string') : IValueSuggestiions => {
+export const useValueSuggestion = (
+  filterField: string,
+  type: 'string' | 'boolean' = 'string'
+): IValueSuggestion => {
   const [suggestedValues, setSuggestedValues] = useState<string[] | boolean[]>([]);
   const [query, setQuery] = useState<string>('');
   const [isLoading, setIsLoading] = useState(true);
   const data = getDataPlugin();
   const indexPattern = useIndexPattern();
 
+  const getValueSuggestions = async (field) => {
+    const result = await data.autocomplete.getValueSuggestions({
+      query,
+      indexPattern: indexPattern as IIndexPattern,
+      field,
+    });
+
+    if (!result.length) {
+      return getCustomValueSuggestion(field);
+    }
```

Comment:
**thought**: I am not sure what is this doing. does it get fixed values? It is getting some prefixed values, if this is the case, maybe we could let the component receive these as props. We should discuss the solution.

Reply:
If the search does not obtain results, use the new helper that I have created to obtain them (the idea is to separate that responsibility from the hook) the helper can be called here or before calling the hook, if we could define it better 👍🏼